plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -950,181 +977,149 @@ public function get_field_from_customer( $key, $customer, $group = '' ) {
-			$key      = str_replace( '/shipping/', '', $key );
-		} else {
-			$meta_key = self::ADDITIONAL_FIELDS_KEY;
-		}
-
-		$meta_data = $object->get_meta( $meta_key, true );
+	private function get_field_from_object( string $key, WC_Data $wc_object, string $group = 'additional' ) {
+		$meta_key = self::get_group_key( $group ) . $key;
 
-		if ( ! is_array( $meta_data ) ) {
-			return '';
-		}
+		$meta_data = $wc_object->get_meta( $meta_key, true );
 
-		if ( ! isset( $meta_data[ $key ] ) ) {
+		if ( '' === $meta_data || null === $meta_data ) {
 			return '';
 		}
 
-		return $meta_data[ $key ];
+		return $meta_data;
 	}
 
 	/**
-	 * Returns an array of all fields values for a given customer.
+	 * Returns an array of all fields values for a given customer in a group.
 	 *
 	 * @param WC_Customer $customer The customer to get the fields for.
+	 * @param string      $group The group to get the fields for (shipping|billing|additional).
 	 * @param bool        $all Whether to return all fields or only the ones that are still registered. Default false.
 	 *
 	 * @return array An array of fields.
 	 */
-	public function get_all_fields_from_customer( $customer, $all = false ) {
-		$meta_data = [
-			'billing'    => [],
-			'shipping'   => [],
-			'additional' => [],
-		];
+	public function get_all_fields_from_customer( WC_Customer $customer, string $group = 'additional', bool $all = false ) {
+		$meta_data = [];
+
+		$prefix = self::get_group_key( $group );
 
 		if ( $customer instanceof WC_Customer ) {
-			$meta_data['billing']    = $customer->get_meta( self::BILLING_FIELDS_KEY, true );
-			$meta_data['shipping']   = $customer->get_meta( self::SHIPPING_FIELDS_KEY, true );
-			$meta_data['additional'] = $customer->get_meta( self::ADDITIONAL_FIELDS_KEY, true );
+			// Get all customer meta regardless of key.
+			$meta = $customer->get_meta_data();
+			foreach ( $meta as $meta_data_object ) {
+				// Check if key starts with a prefix.
+				if ( 0 === \strpos( $meta_data_object->key, $prefix ) ) {
+					$key = \str_replace( $prefix, '', $meta_data_object->key );
+					if ( ! $all && ! $this->is_field( $key ) ) {
+						continue;
+					}
+					$meta_data[ $key ] = $meta_data_object->value;
+				}
+			}
 		}
-		return $this->format_meta_data( $meta_data, $all );
+
+		return $meta_data;
 	}
 
 	/**
 	 * Returns an array of all fields values for a given order.
 	 *
 	 * @param WC_Order $order The order to get the fields for.
+	 * @param string   $group The group to get the fields for (shipping|billing|additional).
 	 * @param bool     $all Whether to return all fields or only the ones that are still registered. Default false.
 	 *
 	 * @return array An array of fields.
 	 */
-	public function get_all_fields_from_order( $order, $all = false ) {
-		$meta_data = [
-			'billing'    => [],
-			'shipping'   => [],
-			'additional' => [],
-		];
+	public function get_all_fields_from_order( WC_Order $order, string $group = 'additional', bool $all = false ) {
```

Comment:
Maybe remove `get_all_fields_from_order` and `get_all_fields_from_customer` and make `get_all_fields_from_object` public.